### PR TITLE
Fix effect bubbles not showing

### PR DESCRIPTION
**Pull request by IvanCraft623**
As of 1.21 bubbles for each effect are displayed separately (it is no longer a mix of all visible effect colors), so it requires that we send all visible effect ids.

## Tests
I tested this PR by doing the following (tick all that apply):
- [x] Playtesting using a Minecraft client (provide screenshots or a video)
- [x] Ensured that the values generated were identical to those generated by BDS.

### CODE REVIEW

File: src/entity/Living.php [hunk 1]
```diff
@@ -887,22 +886,21 @@ protected function sendSpawnPacket(Player $player) : void{
...
+		}
 
-		//TODO: HACK! the client may not be able to identify effects if they are not sorted.
-		sort($visibleEffectIds, SORT_NUMERIC);
+		ksort($visibleEffects, SORT_NUMERIC);
 
 		$effectsData = 0;
-		foreach ($visibleEffectIds as $effectId) {
-			$effectsData = ($effectsData << 7) | ($effectId << 1);
+		foreach (array_slice($visibleEffects, 0, 8, true) as $effectId => $isAmbient) {
```
Comment (dktapps) [inline src/entity/Living.php]: This allocates an extra array, a counter to break would be better

**Review by dktapps — COMMENTED** [review]
Not sure why my review was requested for this. I don't know anything about this new stuff. I only stopped by to leave a code style comment.

**Comment by dries-c** [comment]
Seems like this also resolves the issue where the second time the same effect is issued it won't show up anymore.

**Review by dries-c — DISMISSED** [review]

**Review by ShockedPlot7560 — DISMISSED** [review]

**Comment by IvanCraft623** [comment]
@dries-c @ShockedPlot7560 Re-review :D

**Comment by ShockedPlot7560** [comment]
:D

**Review by ShockedPlot7560 — APPROVED** [review]

**Review by dries-c — APPROVED** [review]

**Pull request merged**